index.php

```diff
@@ -145,93 +145,90 @@
             </div>
         </div>
 
+<?php
+include 'connectdb.php';
+$dishesPerPage = 12;
+$page = isset($_GET['page']) ? (int)$_GET['page'] : 1;
+$offset = ($page - 1) * $dishesPerPage;
+
+$filters = array();
+$params = array();
+
+// Handle search keyword
+if (isset($_GET['keyword']) && !empty($_GET['keyword'])) {
+    $filters[] = "dish_name LIKE :keyword OR description LIKE :keyword";
+    $params[':keyword'] = '%' . $_GET['keyword'] . '%';
+}
+
+// Handle notes filter
+if (isset($_GET['notes']) && $_GET['notes'] != '') {
+    $filters[] = "note = :note";
+    $params[':note'] = $_GET['notes'];
+}
+
+// Handle price filter
+if (isset($_GET['price']) && $_GET['price'] != '') {
+    switch ($_GET['price']) {
+        case 'l10':
+            $filters[] = "price < 10";
+            break;
+        case '1020':
+            $filters[] = "price BETWEEN 10 AND 20";
+            break;
+        // Add cases for other price ranges
+    }
+}
+
+// Handle content filter and dish type filter similarly
+
+// Construct the SQL query
+$sql = "SELECT * FROM menu";
+if (!empty($filters)) {
+    $sql .= " WHERE " . implode(" AND ", $filters);
+}
+$sql .= " LIMIT :offset, :limit";
+
+// Prepare and execute the statement
+$stmt = $conn->prepare($sql);
+$stmt->bindValue(':offset', $offset, PDO::PARAM_INT);
+$stmt->bindValue(':limit', $dishesPerPage, PDO::PARAM_INT);
+foreach ($params as $key => $value) {
+    $stmt->bindValue($key, $value);
+}
+$stmt->execute();
+$dishes = $stmt->fetchAll(PDO::FETCH_ASSOC);
 
+// Fetch total number of dishes
+$totalSql = "SELECT COUNT(*) AS total FROM menu";
+if (!empty($filters)) {
+    $totalSql .= " WHERE " . implode(" AND ", $filters);
+}
+$totalStmt = $conn->prepare($totalSql);
+foreach ($params as $key => $value) {
+    $totalStmt->bindValue($key, $value);
+}
+$totalStmt->execute();
+$total = $totalStmt->fetch(PDO::FETCH_ASSOC)['total'];
+$totalPages = ceil($total / $dishesPerPage);
+?>
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider extracting duplicated pagination logic</b></div>
<div id="fix">

Consider extracting the pagination and filtering logic into a separate function or class to improve code organization and reusability. Similar code appears in `menu.php` and `get_dishes.php`.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -1,1 +1,32 @@
 +class DishPaginator {
 +    private $conn;
 +    private $dishesPerPage;
 +
 +    public function __construct($conn, $dishesPerPage = 12) {
 +        $this->conn = $conn;
 +        $this->dishesPerPage = $dishesPerPage;
 +    }
 +
 +    public function getDishes($page, $filters = [], $params = []) {
 +        $offset = ($page - 1) * $this->dishesPerPage;
 +        
 +        $sql = "SELECT * FROM menu";
 +        if (!empty($filters)) {
 +            $sql .= " WHERE " . implode(" AND ", $filters);
 +        }
 +        $sql .= " LIMIT :offset, :limit";
 +
 +        $stmt = $this->conn->prepare($sql);
 +        $stmt->bindValue(':offset', $offset, PDO::PARAM_INT);
 +        $stmt->bindValue(':limit', $this->dishesPerPage, PDO::PARAM_INT);
 +        foreach ($params as $key => $value) {
 +            $stmt->bindValue($key, $value);
 +        }
 +        $stmt->execute();
 +        return $stmt->fetchAll(PDO::FETCH_ASSOC);
 +    }
 +}

```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Lux-Jason/rms_new/pull/5#issuecomment-2543168959>#f312df</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged